dqc_us_rules/__init__.py

```diff
@@ -12,12 +12,17 @@ def run_checks(val):
     """
     plugin_modules = _plugins_to_run(sys.modules[__name__])
     for plugin in plugin_modules:
-        if ((plugin.__file__ is not None and
-             plugin.__file__.find('__init__.py') == -1 and
-             hasattr(plugin, '__pluginInfo__'))):
-
-            func = plugin.__pluginInfo__['Validate.XBRL.Finally']
-            func(val)
+        try:
+            if ((plugin.__file__ is not None and
+                 plugin.__file__.find('__init__.py') == -1 and
+                 hasattr(plugin, '__pluginInfo__'))):
+
+                func = plugin.__pluginInfo__['Validate.XBRL.Finally']
+                func(val)
+        except Exception:
+            # This is an overly generic error catch, but it will hopefully
+            # be able to be pared down in the future.
+            val.modelXbrl.error(sys.exc_info())
```

Comment:
This should be more than just the exc_info().  It needs to have an error key, and a few other things as well.  I will attempt to find an example of this.